src/Workspaces/Core/Portable/Workspace/Solution/TextLoader.cs

```diff
@@ -31,6 +31,18 @@ public abstract class TextLoader
 
     internal virtual string? FilePath => null;
 
+    /// <summary>
+    /// <see langword="true"/> if the document that holds onto this loader should do so with a strong reference, versus
+    /// a reference that will take the contents of this loader and store them in a recoverable form (e.g. a memory
+    /// mapped file within the <em>same</em> process).  This should be used when the underlying data is already stored
+    /// in a recoverable form somewhere else and it would be wasteful to store another copy.  For example, a document
+    /// that is backed by memory-mapped contents in <em>another</em> process does not need to dump it's content to
+    /// another memory-mapped file in the process it lives in.  It can always recover the text from the original
+    /// process.
+    /// </summary>
+    internal virtual bool AlwaysHoldStrongly
+        => false;
```

Comment:
most loaders are 'one and done' in terms of loading.  Once they load the contents, you have to hold onto the contents yourself so that they do not change between the first time calling the loader and the next time.  We do htat in roslyn by loading the contents, tehn saving them to MMFs to give them snapshot semantics.

There's no need for this though with the TextLoaders we create in OOP that wrap the MMFs for files created on the host side.  These *can* safey be reloaded as they will give back the exact same contents on reload.